code/game/objects/structures/aliens.dm

```diff
@@ -254,24 +258,25 @@
 	status = GROWN
 	AddComponent(/datum/component/proximity_monitor)
 
-/obj/structure/alien/egg/proc/Burst(kill = TRUE)	//drops and kills the hugger if any is remaining
+/obj/structure/alien/egg/proc/Burst(kill)	///Need to carry the kill from Burst() to Hatch()
 	if(status == GROWN || status == GROWING)
 		icon_state = "egg_hatched"
 		flick("egg_opening", src)
 		status = BURSTING
 		qdel(GetComponent(/datum/component/proximity_monitor))
-		spawn(15)
-			status = BURST
-			var/obj/item/clothing/mask/facehugger/child = GetFacehugger()
-			if(child)
-				child.loc = get_turf(src)
-				if(kill && istype(child))
-					child.Die()
-				else
-					for(var/mob/M in range(1,src))
-						if(CanHug(M))
-							child.Attach(M)
-							break
+		addtimer(CALLBACK(src, /obj/structure/alien/egg/.proc/Hatch), 1.5 SECONDS)
+
+/obj/structure/alien/egg/proc/Hatch(kill) ///We now check HOW the hugger is hatching, kill carried from Burst() and obj_break()
+	status = BURST
+	var/obj/item/clothing/mask/facehugger/child = GetFacehugger()
+	if(child)
+		child.loc = get_turf(src)
+	if(kill && istype(child))
```

Comment:
will `GetFacehugger()` ever return an obj that isn't the exact type you're looking for?

Reply:
I don't think so, I can add support for people adding a different item through varediting though

Reply:
I'm just asking because if you never expect for `GetFacehugger()` to return anything other than a `obj/item/clothing/mask/facehugger/child` we don't need to use an istype here

Reply:
All reviews addressed in latest commit